File: mRemoteV1/CS/Tools/PuttyTypeDetector.cs
using System.Collections.Generic;
using System;
using AxWFICALib;
using System.Drawing;
using System.Diagnostics;
using System.Data;
using AxMSTSCLib;
using Microsoft.VisualBasic;
using System.Collections;
using System.Windows.Forms;
using mRemoteNG.Connection.Protocol;


namespace mRemoteNG.Tools
{
	public class PuttyTypeDetector
	{
		public static PuttyType GetPuttyType()
		{
			return GetPuttyType(PuttyBase.PuttyPath);
		}
			
		public static PuttyType GetPuttyType(string filename)
		{
			if (IsPuttyNg(filename))
			{
				return PuttyType.PuttyNg;
			}
			if (IsKitty(filename))
			{
				return PuttyType.Kitty;
			}
			if (IsXming(filename))
			{
				return PuttyType.Xming;
			}
				
			// Check this last
			if (IsPutty(filename))
			{
				return PuttyType.Putty;
			}
				
			return PuttyType.Unknown;
		}
			
		private static bool IsPutty(string filename)
		{
			bool result = false;
			try
			{
				result = System.Convert.ToBoolean(FileVersionInfo.GetVersionInfo(filename).InternalName.Contains("PuTTY"));
			}
			catch
			{
				result = false;
			}
			return result;
		}
			
		private static bool IsPuttyNg(string filename)
		{
			bool result = false;
			try
			{
				result = System.Convert.ToBoolean(FileVersionInfo.GetVersionInfo(filename).InternalName.Contains("PuTTYNG"));
			}
			catch
			{
				result = false;
			}
			return result;
		}
			
		private static bool IsKitty(string filename)
		{
			bool result = false;
			try
			{
				result = System.Convert.ToBoolean(FileVersionInfo.GetVersionInfo(filename).InternalName.Contains("PuTTY") && FileVersionInfo.GetVersionInfo(filename).Comments.Contains("KiTTY"));
			}
			catch
			{
				result = false;
			}
			return result;
		}
			
		private static bool IsXming(string filename)
		{
			bool result = false;
			try
			{
				result = System.Convert.ToBoolean(FileVersionInfo.GetVersionInfo(filename).InternalName.Contains("PuTTY") && FileVersionInfo.GetVersionInfo(filename).ProductVersion.Contains("Xming"));
			}
			catch
			{
				result = false;
			}
			return result;
		}
			
		public enum PuttyType
		{
			Unknown = 0,
			Putty,
			PuttyNg,
			Kitty,
			Xming
		}
	}
}

File: mRemoteV1/Tools/PuttyTypeDetector.vb
Imports mRemote3G.Connection.Protocol

Namespace Tools
    Public Class PuttyTypeDetector
        Public Shared Function GetPuttyType() As PuttyType
            Return GetPuttyType(PuttyBase.PuttyPath)
        End Function

        Public Shared Function GetPuttyType(filename As String) As PuttyType
            If IsPuttyNg(filename) Then Return PuttyType.PuttyNg
            If IsKitty(filename) Then Return PuttyType.Kitty
            If IsXming(filename) Then Return PuttyType.Xming

            ' Check this last
            If IsPutty(filename) Then Return PuttyType.Putty

            Return PuttyType.Unknown
        End Function

        Private Shared Function IsPutty(filename As String) As Boolean
            Dim result As Boolean
            Try
                result = FileVersionInfo.GetVersionInfo(filename).InternalName.Contains("PuTTY")
            Catch
                result = False
            End Try
            Return result
        End Function

        Private Shared Function IsPuttyNg(filename As String) As Boolean
            Dim result As Boolean
            Try
                result = FileVersionInfo.GetVersionInfo(filename).InternalName.Contains("PuTTYNG")
            Catch
                result = False
            End Try
            Return result
        End Function

        Private Shared Function IsKitty(filename As String) As Boolean
            Dim result As Boolean
            Try
                result = FileVersionInfo.GetVersionInfo(filename).InternalName.Contains("PuTTY") _
                         And FileVersionInfo.GetVersionInfo(filename).Comments.Contains("KiTTY")
            Catch
                result = False
            End Try
            Return result
        End Function

        Private Shared Function IsXming(filename As String) As Boolean
            Dim result As Boolean
            Try
                result = FileVersionInfo.GetVersionInfo(filename).InternalName.Contains("PuTTY") _
                         And FileVersionInfo.GetVersionInfo(filename).ProductVersion.Contains("Xming")
            Catch
                result = False
            End Try
            Return result
        End Function

        Public Enum PuttyType
            Unknown = 0
            Putty
            PuttyNg
            Kitty
            Xming
        End Enum

        Private Sub New()
        End Sub
    End Class
End Namespace